src/System.Linq/src/System/Linq/Enumerable.cs

```diff
@@ -703,11 +709,13 @@ public static IEnumerable<TSource> SkipWhile<TSource>(this IEnumerable<TSource>
 
         private static IEnumerable<TSource> SkipWhileIterator<TSource>(IEnumerable<TSource> source, Func<TSource, bool> predicate)
         {
-            bool yielding = false;
-            foreach (TSource element in source)
+            using (IEnumerator<TSource> e = source.GetEnumerator())
             {
-                if (!yielding && !predicate(element)) yielding = true;
-                if (yielding) yield return element;
+                while (e.MoveNext())
+                    if (!predicate(element)) break;
+                do
+                    yield return element;
```

Comment:
@stephentoub Whoops, sorry. I was working on adding tests to `SkipWhile` as you were writing this, and now I've fixed the implementation as well when I got error results from the test. I'll work on adding more tests as I go along.